File: app/java/win/code/src/manager/GTest.java
//===============================================
package manager;
//===============================================
public class GTest extends GObject {  
    //===============================================
    public GTest() {  

    }  
    //===============================================
    public void run(String[] _args) {
        String lMethod = "";
        if(_args.length > 1) lMethod = _args[1];
        
        if(lMethod.equals("")) {
            m_logs.addError("La méthode est obligatoire.");
        }
        else if(lMethod.equals("xml")) {
            runXml(_args);
        }
        else if(lMethod.equals("code")) {
            runCode(_args);
        }
        else if(lMethod.equals("log")) {
            runLog(_args);
        }
        else if(lMethod.equals("socket")) {
            runSocket(_args);
        }
        else if(lMethod.equals("thread")) {
            runThread(_args);
        }
        else {
            m_logs.addError("La méthode est inconnue.");
        }
    }        
    //===============================================
    public void runXml(String[] _args) {
        GXml lDom = new GXml();
        GXml lDomC = new GXml();
        String lData = "";
        
        // createDoc - addObj - addValue - toString - setNode - print
        lDom.createDoc();
        lDomC.setNode(lDom.addObj("datas"));
        lDomC.setNode(lDomC.addObj("data"));
        lDomC.addValue("code", "logs");
        lData = lDom.toString();
        lDom.print();
        
        // loadXml
        lDom.loadXml(lData);
        lDomC.setNode(lDom.getNode("/rdv/datas"));
        lDomC.printNode();

        // loadXml
        lDom.createDoc();
        lDom.createNode("/rdv/datas/data");
        lDom.createNode("/rdv/datas/data");
        lDom.print();
    }
    //===============================================
    public void runCode(String[] _args) {
        GCode lDom = new GCode();
        
        // createDatas - createCode - addData
        lDom.createDoc();
        lDom.createDatas();
        lDom.createDatas();
        lDom.createCode("logs");
        lDom.createCode("logs");
        lDom.addData("logs", "type", "error");
        lDom.addData("logs", "side", "server_java");
        lDom.addData("logs", "msg", "Le serveur n'est pas disponible.");
        lDom.addData("logs", "type", "log");
        lDom.print();
    }
    //===============================================
    public void runLog(String[] _args) {
        GLog lLog = new GLog();
        GLog lLogC = new GLog();
        
        // addError - createCode - addData - print
        lLog.addError("Le serveur n'est pas disponible.");
        lLog.addLog("Le module a été chargé.");
        lLog.addData("La résolution de l'appareil est : 970 x 1250.");
        lLog.loadFromMap(3);
        lLog.print();
        
        // serialize - deserialize
        lLogC.deserialize(lLog.serialize());
        lLogC.print();
    }
    //===============================================
    public void runSocket(String[] _args) {
        GCalculator lCalculator = new GCalculator();
        lCalculator.setExpression("22/7");
        lCalculator.runCalculator();
        m_logs.addLogs(lCalculator.getLogs());
        System.out.print(String.format("%s\n", lCalculator.serialize()));
    }
    //===============================================
    public void runThread(String[] _args) {
        GThread lThread = new GThread();
        lThread.start();
    }
    //===============================================
}
//===============================================

File: app/php/client/code/src/php/class/GTest.php
<?php   
//===============================================
class GTest extends GObject {
    //===============================================
    public function __construct() {
        parent::__construct();
    }
    //===============================================
    public function run($_module, $_method) {
        if($_method == "xml") {
            $this->runXml($_module, $_method);
        }
        else if($_method == "code") {
            $this->runCode($_module, $_method);
        }
        else if($_method == "log") {
            $this->runLog($_module, $_method);
        }
        else if($_method == "curl_client") {
            $this->runCurlClient($_module, $_method);
        }
        else if($_method == "socket_client") {
            $this->runSocketClient($_module, $_method);
        }
    }
    //===============================================
    public function runJs($_module, $_method) {
        echo sprintf("runJs...<br>\n");
        echo sprintf("<script>call_server('test', '%s');</script>\n", $_method);
    }
    //===============================================
    public function runXml($_module, $_method) {
        echo sprintf("runXml...<br>\n");
        $lDom = new GXml();
        $lDomC1 = new GXml();
        $lDomC2 = new GXml();
        
        // createDoc - setNode - addObj - addData - toString
        $lDom->createDoc();
        $lDomC1->setNode($lDom->addObj($lDom, "datas"));
        $lDomC2->setNode($lDomC1->addObj($lDom, "data"));
        $lDomC2->setNode($lDomC2->addData($lDom, "code", "logs"));
        $lDomC2->setNode($lDomC2->addData($lDom, "type", "error"));
        $lDomC2->setNode($lDomC2->addData($lDom, "msg", "La connexion a échoué."));
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
        
        // createNode
        $lDom->createDoc();
        $lDomC1->setNode($lDom->createNode($lDom, "/rdv/datas/data"));
        $lDomC1->setNode($lDom->createNode($lDom, "/rdv/datas/data"));
        $lDomC2->setNode($lDomC1->createNode($lDom, "rdv/datas/data"));
        $lDomC2->setNode($lDomC1->createNode($lDom, "rdv/datas/data"));
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
        
        // loadXml
        $lDom->loadXml($lDom->toString());
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
        
        // loadXml
        $lData = "";
        $lData .= "<rdv>";
        $lData .= "<datas>";
        $lData .= "<data>";
        $lData .= "<code>logs</code>";
        $lData .= "<type>error</type>";
        $lData .= "<msg>La connexion a échoué.</msg>";
        $lData .= "</data>";
        $lData .= "</datas>";
        $lData .= "</rdv>";
        $lDom->loadXml($lData);
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
        
        // toNode
        $lDom->createDoc();
        $lDomC1->setNode($lDom->addObj($lDom, "datas"));
        $lDomC2->setNode($lDomC1->addObj($lDom, "data"));
        $lDomC2->setNode($lDomC2->addData($lDom, "code", "logs"));
        $lDomC2->setNode($lDomC2->addData($lDom, "type", "error"));
        $lDomC2->setNode($lDomC2->addData($lDom, "msg", "La connexion a échoué."));
        echo sprintf("<xmp>%s</xmp>\n", $lDomC1->toNode($lDom));
        
        // loadNode
        $lData = "";
        $lData .= "<datas>";
        $lData .= "<data>";
        $lData .= "<code>logs</code>";
        $lData .= "<type>error</type>";
        $lData .= "<msg>La connexion a échoué.</msg>";
        $lData .= "</data>";
        $lData .= "</datas>";
        $lDom->createDoc();
        $lDomC1->setNode($lDom->createNode($lDom, "/rdv/datas"));
        $lDomC1->loadNode($lDom, $lData);
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
    }
    //===============================================
    public function runCode($_module, $_method) {
        echo sprintf("runCode...<br>\n");
        
        // createDoc - createDatas
        $lDom = new GCode();
        $lDom->createDoc();
        $lDom->createDatas();
        $lDom->createDatas();
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
        
        // createCode
        $lDom = new GCode();
        $lDom->createDoc();
        $lDom->createCode("logs");
        $lDom->createCode("logs");
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
        
        // serialize
        $lLog = new GLog();
        $lLog->addError("Le serveur n'est pas disponible.");
        $lLog->addData("La taille de la fenêtre est: 1000.");
        $lLog->addLog("Le chargement du module est terminé.");
        $lLog->loadFromMap(3);
        echo sprintf("<xmp>%s</xmp>\n", $lLog->serialize());
        
        // addData - toDatas
        $lDom = new GCode();
        $lDom->createDoc();
        $lDom->addData("manager", "module", "logs");
        $lDom->addData("manager", "method", "load_logs");
        $lDom->addData("manager", "module", "logs_2");
        $lDom->addData("manager", "method", "load_logs_2");
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
        
        // loadData
        $lDom = new GCode();
        $lDom->createDoc();
        $lDom->addData("manager", "module", "logs");
        $lDom->addData("manager", "method", "load_logs");
        $lDom->loadData($lLog->serialize());
        echo sprintf("<xmp>%s</xmp>\n", $lDom->toString());
        
        // deserialize
        $lLog2 = new GLog();
        $lLog2->deserialize($lLog->serialize());
        echo sprintf("<xmp>%s</xmp>\n", $lLog2->serialize());
        
    }
    //===============================================
    public function runLog($_module, $_method) {
        echo sprintf("runLog...<br>\n");
        
        // addError - addData - addLog - addLogs
        $lLog = new GLog();
        $lLog->addError("Le serveur n'est pas disponible.");
        $lLog->addData("La taille de la fenêtre est: 1000.");
        $lLog->addLog("Le chargement du module est terminé.");
        $this->m_logs->addLogs($lLog);
    }
    //===============================================
    public function runCurlClient($_module, $_method) {
        echo sprintf("runCurlClient...<br>\n");
    }
    //===============================================
    public function runSocketClient($_module, $_method) {
        echo sprintf("runSocketClient...<br>\n");
        
        // callFacade
        $lClient = new GSocket();
        $lData = $lClient->callFacade("logs", "save_logs");
        $this->m_logs->deserialize($lData);
        $this->m_logs->addLogs($lClient->getLogs());
    }
    //===============================================
 }
//===============================================
?>
